types/decimal.go

```diff
@@ -1,28 +1,91 @@
 package types
 
 import (
+	"cmp"
 	"encoding/json"
 	"errors"
 	"fmt"
+	"math"
 	"strconv"
 	"unicode"
+
+	"golang.org/x/exp/constraints"
 )
 
+// decimalPrecision is the precision of a Decimal.
+const decimalPrecision = 10000
+
+var DecimalMax = Decimal{value: math.MaxInt64}
```

Comment:
Do these really need to be public?

Reply:
or at least `const`?

Reply:
The tests can just assert that they got an error back and not be particular about the type of error.  Or the errors could live in an internal package that the tests can access as well as the package can access.

I suppose the harm is everything we make public must be there for all eternity as of 1.0, so unless there is an argument for making it public, it's better to keep it private.

Reply:
Oh sorry, I was talking about the public errors, but the same gist applies here WRT internal package.